pkgs/applications/graphics/upscayl/default.nix

```diff
@@ -0,0 +1,38 @@
+{
+  appimageTools,
+  fetchurl,
+  lib,
+}: let
+  pname = "upscayl";
+  version = "2.5.5";
+  name = "${pname}-${version}";
+  src = fetchurl {
+    url = "https://github.com/upscayl/upscayl/releases/download/v${version}/upscayl-${version}-linux.AppImage";
+    hash = "sha256-qpLxOGphR9iHvtb8AZZaMict/g8wLkL7Dhr4mt3LZdk=";
+  };
+  appimageContents = appimageTools.extractType2 {
+    inherit name src;
+  };
+in
+  appimageTools.wrapType2 rec {
+    inherit name src;
+
+    extraPkgs = pkgs: with pkgs; [vulkan-headers vulkan-loader];
+
+    extraInstallCommands = ''
+      mkdir -p $out/share/{applications,pixmaps}
+
+      cp ${appimageContents}/${pname}.desktop $out/share/applications/${pname}.desktop
+      cp ${appimageContents}/${pname}.png $out/share/pixmaps/${pname}.png
+
+      substituteInPlace $out/share/applications/${pname}.desktop \
+        --replace 'Exec=AppRun --no-sandbox %U' 'Exec=${pname}'
+    '';
+
+    meta = with lib; {
```

Comment:
The `license` is missing.

Reply:
Set the license to `licenses.agpl3Only;` for now and will create an issue to confirm if the author wants to support only that version or future versions of the AGPL license.